conda/core/envs_manager.py

```diff
@@ -39,6 +39,7 @@ def _clean_environments_txt(environments_txt_file, remove_location=None):
         return ()
 
     environments_txt_lines = list(yield_lines(environments_txt_file))
+    original_environments_txt_hash = hash(tuple(environments_txt_file))
```

Comment:
Should these be sorted?

Reply:
Thanks for looking at this PR.  I don't think it matters that they're sorted.  What matters is that it has the same order as `hash(real_prefixes)` in line 52.  However, I think I should add a test that ensures this is all working correctly.

Reply:
But does order matter in environments.txt? If not, and the goal is to eliminate unecessary rewrites, then it would seem to me that we should be doing a comparison that matters.

Also, I know a hash is very, very unlikely to let false equivalencies pass through, but why take a chance? What benefit is there to avoiding a full comparison?